Mathlib/Analysis/Normed/Group/Basic.lean

```diff
@@ -1406,6 +1387,42 @@ theorem SeminormedGroup.uniformCauchySeqOn_iff_tendstoUniformlyOn_one {f : ι 
 
 end SeminormedGroup
 
+namespace Mathlib.Meta.Positivity
+
+open Lean Meta Qq Function
+
+/-- Extension for the `positivity` tactic: multiplicative norms are nonnegative, via
+`norm_nonneg'`. -/
+@[positivity Norm.norm _]
+def evalMulNorm : PositivityExt where eval {u α} _ _ e := do
+  match u, α, e with
+  | 0, ~q(ℝ), ~q(@Norm.norm $E $_n $a) =>
+    let _sng ← synthInstanceQ q(SeminormedGroup $E)
+    assumeInstancesCommute
```

Comment:
I'm not sure when to use what - what's the guidelines?

Reply:
`assumeInstancesCommute` essentially means "either I know these must always be defeq, or I am happy to produce a totally cryptic error in the kernel if they are not". `assertInstancesCommute` means "in rare cases these won't be defeq, so I will spend a little extra work every time checking, so that I can produce a better error".